.github/scripts/update_schutzfile.py

```diff
@@ -0,0 +1,38 @@
+#!/bin/python3
+
+import json
+import os
+import re
+
+repo_files = os.listdir("/repo/")
+singletons = []
+# Get a list of all repositories that contain 'singleton'
+for repo_file in repo_files:
+    with open(os.path.join("repo", repo_file), "r") as f:
+        data = json.load(f)
+    if "singleton" in data.keys():
+        singletons.append(data["snapshot-id"])
+
+with open(os.path.join(os.environ.get("REPO"), "Schutzfile"), "r") as f:
+    data = json.load(f)
+
+urls = []
+arches = ["x86_64", "aarch64"]
+# Replace snapshot SUFFIX in repositories that don't have 'singleton' present
+for key in data.keys():
+    if "repos" in data[key].keys():
+        for i, _ in enumerate(data[key]["repos"]):
```

Comment:
This can indeed be `for repo in data[key]["repos"]` because `repo` is a dict and thus it will be a reference (not a copy).

See https://www.online-python.com/jr9hdI0fFw

Reply:
Yeah ok got it working with some adjustments of the following code